velox/functions/prestosql/tests/StringFunctionsTest.cpp

```diff
@@ -1131,6 +1136,90 @@ TEST_F(StringFunctionsTest, codePoint) {
   }
 }
 
+std::optional<int64_t> StringFunctionsTest::levenshteinDistance(
+    const std::string left,
+    const std::string right) {
+  auto leftVector = makeFlatVector<StringView>({StringView(left)});
+  auto rightVector = makeFlatVector<StringView>({StringView(right)});
+  return evaluateOnce<int64_t>(
+      "levenshtein_distance(c0, c1)", makeRowVector({leftVector, rightVector}));
+}
+
+TEST_F(StringFunctionsTest, levenshteinDistance) {
+  // All ASCII characters
+  EXPECT_EQ(levenshteinDistance("", "").value(), 0);
+  EXPECT_EQ(levenshteinDistance("", "hello").value(), 5);
+  EXPECT_EQ(levenshteinDistance("hello", "").value(), 5);
+  EXPECT_EQ(levenshteinDistance("hello", "hello").value(), 0);
+  EXPECT_EQ(levenshteinDistance("hello", "olleh").value(), 4);
+  EXPECT_EQ(levenshteinDistance("hello world", "hello").value(), 6);
+  EXPECT_EQ(levenshteinDistance("hello", "hello world").value(), 6);
+  EXPECT_EQ(levenshteinDistance("hello world", "hel wold").value(), 3);
+  EXPECT_EQ(levenshteinDistance("hello world", "hellq wodld").value(), 2);
+  EXPECT_EQ(levenshteinDistance("hello word", "dello world").value(), 2);
+  EXPECT_EQ(
+      levenshteinDistance("hello", std::string(100000, 'h')).value(), 99999);
+  EXPECT_EQ(
+      levenshteinDistance(std::string(100000, 'l'), "hello").value(), 99998);
+  EXPECT_EQ(
+      levenshteinDistance(std::string(1000001, 'h'), "").value(), 1000001);
+  EXPECT_EQ(
+      levenshteinDistance("", std::string(1000001, 'h')).value(), 1000001);
+
+  // Unicode
+  EXPECT_EQ(
+      levenshteinDistance("hello na\u00EFve world", "hello naive world")
+          .value(),
+      1);
+  EXPECT_EQ(
+      levenshteinDistance("hello na\u00EFve world", "hello na:ive world")
+          .value(),
+      2);
+  EXPECT_EQ(
+      levenshteinDistance(
+          "\u4FE1\u5FF5,\u7231,\u5E0C\u671B",
+          "\u4FE1\u4EF0,\u7231,\u5E0C\u671B")
+          .value(),
```

Comment:
change the signature of levenshteinDistance to avoid having to call .value() again and again